python/tank/context.py

```diff
@@ -624,37 +623,105 @@ def _fields_from_template_tree(self, template, fields, entities):
         path_cache = PathCache(self.__tk)
 
         # Step 3 - walk templates from the root down,
-        # for each template, get all paths we have stored in the database
-        # and find any fields we can for it
+        # for each template, get all paths we have stored in the database and find any fields we can for it, making 
+        # sure that none of the found fields conflict with the list of entities provided to this method
         try:
             # build up a list of fields as we go so that each level matches
-            # at least the fields from the previous level
+            # at least the fields from all previous levels
             found_fields = {}
 
-            for cur_template in templates:
-                for key in cur_template.keys.values():
+            for template in templates:
+                for key_name in template.keys.keys():
                     # If we don't already have a value, look for it
-                    if fields.get(key.name) is not None:
+                    if key_name in all_fields:
                         # already have value so skip:
-                        found_fields[key.name] = fields[key.name]
                         continue
-                    
-                    # only care about entities as this is what we'll look for in the path cache:
-                    entity = entities.get(key.name)
-                    if entity:
-                        # context contains an entity for this Shotgun entity type!
-                        temp_fields = _values_from_path_cache(entity, cur_template, path_cache, 
-                                                              required_fields=found_fields)
-                        # make sure the next iteration finds the same fields: 
-                        found_fields.update(temp_fields)
-            
-            # update the list of fields with all the ones we found:
-            fields.update(found_fields)
 
-        finally:    
+                    if key_name not in entities:
+                        # key doesn't represent an entity so skip
```

Comment:
The entities dict is really the *context* dictionary right? I think maybe we should rename it to that?
Am I right thinking that there may be a case where you have the template `sequences/{Sequence}/{Shot}` but the `entitites` dict does not contain `Sequence` because the context is set to be the Shot? Maybe the comment just needs updating? Sorry if I am on the wrong track here :)